news/_posts/2020-09-30-how-to-write-a-new-ui-test.md

```diff
@@ -0,0 +1,376 @@
+  await expect(numberOfOrders).to.be.above(0);
+});
+
+it('should filter order by customer name', async function () {
+  // Action
+  await ordersPage.filterOrders(
+    page,
+    'input',
+    'customer',
+    DefaultAccount.lastName,
+  );
+
+  // Expected result
+  // Check that we have at least 1 order for the customer named "J. DOE"
+  const numberOfOrders = await ordersPage.getNumberOfElementInGrid(page);
+  await expect(numberOfOrders).to.be.at.least(1);
+});
+
+it('should check customer link', async function () {
+  // Action
+  page = await ordersPage.viewCustomer(page, 1);
+
+  // Expected result
+  // Verify that the current page is "view customer" by checking the title 'View information about J. DOE'
+  const pageTitle = await viweCustomerPage.getPageTitle(page);
+  await expect(pageTitle).to
+    .contains(`${viweCustomerPage.pageTitle} ${DefaultAccount.firstName[0]}. ${DefaultAccount.lastName}`);
+});
+```
+
+### Implementing a missing method
+
+For this scenario, we used mostly existing functions like *filterOrders()*. You can check for existing functions in the `pages` directory. 
+
+```js
+  /**
+   * Filter Orders
+   * @param page
+   * @param filterType
+   * @param filterBy
+   * @param value
+   * @return {Promise<void>}
+   */
+  async filterOrders(page, filterType, filterBy, value = '') {
+    switch (filterType) {
+      case 'input':
+        await this.setValue(page, this.filterColumn(filterBy), value.toString());
+        break;
+
+      case 'select':
+        await this.selectByVisibleText(page, this.filterColumn(filterBy), value);
+        break;
+
+      default:
+      throw new Error(`${filterBy} was not found as a column filter.`);
+    }
+    // click on search
+    await this.clickAndWaitForNavigation(page, this.filterSearchButton);
+  }
+```
+
+But there's a chance that a method you need is not implemented yet. Here it's the case for the *viewCustomer()* function.
+ 
+```js
+/**
+* Click on customer link to open view page in a new tab
+* @param page
+* @param row
+* @return {Promise<*>}, new browser tab to work with
+*/
+viewCustomer(page, row) {
+  return this.openLinkWithTargetBlank(
+    page,
+    `${this.tableColumn(row, 'customer')} a`,
+    this.userProfileIcon,
+  );
+}
+```
+
+## VI. Running your scenario
+
+After finishing writing your scenario, implementing missing functions... it's time to test your test ! 
+
+```shell script
+TEST_PATH="functional/BO/02_orders/01_orders/08_viewCustomer" URL_FO=shopUrl/ npm run specific-test
+
+ View customer from orders page
+    ✓ should login in BO
+    ✓ should go to orders page
+    ✓ should reset all filters
+    ✓ should filter order by customer name
+    ✓ should check customer link
+ 5 passing (19s)
+```
+
+## VII. Adding steps Identifiers
+
+Steps identifiers help us track failing steps between different runs thanks to the [nightly](https://nightly.prestashop.com/) algorithm.
```

Comment:
```suggestion
Steps identifiers help to track failing steps between different runs thanks to the [nightly](https://nightly.prestashop.com/) algorithm.
```

if possible, avoid the use of "us" or "we" when it's not clear. the target is that everyone can be involved in the tests, not just you and your team ;-)

Reply:
You can remove the "us" altogether.

```suggestion
Steps identifiers help track failing steps between different runs thanks to the [nightly](https://nightly.prestashop.com/) algorithm.
```